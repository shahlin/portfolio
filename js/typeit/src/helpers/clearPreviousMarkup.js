export default (element, isInput = false) => {
  if (isInput) {
    element.value = "";
    return;
  }

  [].slice.call(element.childNodes).forEach(node => {
    if (node.classList === undefined) return;

    if (node.classList.contains("ti-wrapper")) {
      element.innerHTML = "";
    }
  });
};
