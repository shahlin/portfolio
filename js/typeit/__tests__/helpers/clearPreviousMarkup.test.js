import clearPreviousMarkup from "../../src/helpers/clearPreviousMarkup";

test("Should remove .ti-wrapper markup.", () => {
  document.body.innerHTML = `
    <h3 id="example1" data-typeitid="arbitrary-id">
      <span class="ti-wrapper">
        <span class="ti-container">hello!</span>
        <span class="ti-cursor">|</span>
      </span>
    </h3>`;

  clearPreviousMarkup(document.getElementById("example1"));

  expect(true).toMatchSnapshot();
});

test("Should leave other markup alone.", () => {
  document.body.innerHTML = `
    <h3 id="example1" data-typeitid="arbitrary-id">
      This should be left alone.
    </h3>`;

  clearPreviousMarkup(document.getElementById("example1"));

  expect(true).toMatchSnapshot();
});
