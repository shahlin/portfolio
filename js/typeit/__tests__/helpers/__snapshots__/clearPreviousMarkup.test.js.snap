// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should leave other markup alone. 1`] = `true`;

exports[`Should remove .ti-wrapper markup. 1`] = `true`;
